torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1074,6 +1279,19 @@ void ProcessTimeSeriesNode(Node* n) {
   }
 }
 
+void ProcessUnsqueezeNode(Node* n) {
+  TensorTypePtr output_type = n->output(0)->type()->cast<TensorType>();
+  if (output_type == nullptr) {
+    return;
+  }
+  if (output_type->dim().has_value() && output_type->dim().value() == 1 &&
+      ConstantValueMap::HasShapeValue(n->input(0)->debugName())) {
+    auto shape_value =
+        ConstantValueMap::GetShapeValue(n->input(0)->debugName()).value();
+    ConstantValueMap::SetShapeValue(n->output()->debugName(), shape_value);
```

Comment:
why is `ShapeValue` for output the same as input?

Reply:
See the condition `output_type->dim().value() == 1`. The case we consider here is the `Unsqueeze` node with output dim=1, meaning that its input is a scalar. When the scalar represents the shape value, it is essentially the same as  the shape value when it gets unsqueezed.